### PR TITLE
chore: bump to Angular 5.2.x

### PR DESCRIPTION
**Pull request by crisbeto**
Bumps to the latest minor version of Angular now that a compilation error, that was blocking us previously, has been resolved. Also fixes a couple of AoT errors in the demo app.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,11 +41,11 @@
   },
   "devDependencies": {
     "@angular/bazel": "~5.1.1",
```
Comment (crisbeto) [inline package.json]: @jelbourn I didn't bump this one since it ended up introducing a build error in the Bazel check.

Comment (jelbourn) [reply package.json]: What was the error? Theoretically the angular/bazel version should stick to the rest of the packages

Comment (crisbeto) [reply package.json]: Here's what I got. It doesn't look like something from our code:

```
Loading: 0 packages loaded

Analyzing: 138 targets (56 packages loaded)

ERROR: /home/circleci/.cache/bazel/_bazel_circleci/9ce5c2144ecf75d11717c0aa41e45a8d/external/angular/src/ngc-wrapped/BUILD.bazel:18:1: no such package '@build_bazel_rules_typescript_tsc_wrapped_deps//': The repository could not be resolved and referenced by '@angular//src/ngc-wrapped:ngc-wrapped_bin'

ERROR: Analysis of target '//src/lib/datepicker:datepicker' failed; build aborted: no such package '@build_bazel_rules_typescript_tsc_wrapped_deps//': The repository could not be resolved

INFO: Elapsed time: 1.525s

FAILED: Build did NOT complete successfully (61 packages loaded)

Exited with code 1
```

Comment (jelbourn) [reply package.json]: cc @alexeagle 

It sounds like rules_typescript also needs to be updated alongside Angular

Comment (alexeagle) [reply package.json]: right, https://github.com/alexeagle/angular-bazel-example/blob/master/WORKSPACE is an example, that repo uses Angular 5.2.3 (including `@angular/bazel`)

Comment (crisbeto) [reply package.json]: It needed a couple of extra tweaks, but it good to go now @jelbourn.

**Pull request merged**